go/apps/ctrl/services/deployment/create_deployment.go

```diff
@@ -138,16 +123,21 @@ func (s *Service) CreateDeployment(
 
 	s.logger.Info("starting deployment workflow for deployment",
 		"deployment_id", deploymentID,
-		"workspace_id", req.Msg.GetWorkspaceId(),
+		"workspace_id", workspaceID,
 		"project_id", req.Msg.GetProjectId(),
 		"environment", env.ID,
 	)
 
 	// Start the deployment workflow directly
+	keyspaceID := req.Msg.GetKeyspaceId()
+	var keyAuthID *string
+	if keyspaceID != "" {
+		keyAuthID = &keyspaceID
+	}
```

Comment:
this is really just a nit, but it'd be nice if I could just use my api id when deploying instead of having to copy the key auth id from the db // url param

You can just disregard this if it works with keyAuthId in your workflow

Reply:
It doesn't?, the control plane can just lookup the keyAuthId from the apiId.

Requires 0 change on the gateway